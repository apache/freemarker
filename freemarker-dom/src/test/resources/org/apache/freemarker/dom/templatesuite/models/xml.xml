<?xml version="1.0"?>
<!--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<?firstPi customKey="something"?>
<?secondPi secondPiData?>
<ns1:root xmlns:ns1="http://www.foo.com/ns1/">
  <ns1:e1 a1="v1" a2="v2">
    <ns1:e11>text1</ns1:e11>
    <ns1:e12><![CDATA[cdata-section1]]></ns1:e12>
  </ns1:e1>
  <ns1:e2>
    <ns1:e11>text2</ns1:e11>
    <ns1:e12><![CDATA[cdata-section2<&]]></ns1:e12>
  </ns1:e2>
</ns1:root>
