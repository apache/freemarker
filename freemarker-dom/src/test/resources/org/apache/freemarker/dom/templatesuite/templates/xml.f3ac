<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<#-- test processing instructions -->
<#global PIs = doc._content._ftype("p")>
<#list PIs as pi>
  ${pi}
  ${pi["@target"]._text}
  ${pi["@data"]._text}
</#list>
${PIs?size}
<#global firstPi = PIs[0]>
${firstPi._type}
${firstPi["@customKey"]}
${doc._registerNamespace("ns", "http://www.foo.com/ns1/")}
${doc._descendant["ns:e11"]}
${doc._descendant["ns:e12"]}
<#global docRoot = doc["ns:root"]>
${docRoot["ns:e1"]}
${doc("//ns:e11")}
${docRoot["ns:e1"]["@a1"]._name}
${docRoot["ns:e1"]["@a2"]._text}
${docRoot._children._parent._name}
${docRoot._children._parent._unique._name}
<#list doc._descendant as d>
  ${d._name}
</#list>
<#list doc._descendant._ancestorOrSelf as d>
  ${d._name}
</#list>
${docRoot["ns:e2"]["ns:e12"]._text}
${docRoot["ns:e2"]["ns:e12"]._plaintext}
