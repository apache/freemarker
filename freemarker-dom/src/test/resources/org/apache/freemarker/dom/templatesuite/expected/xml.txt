/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
  <?firstPi customKey="something"?>
  firstPi
  customKey="something"
  <?secondPi secondPiData?>
  secondPi
  secondPiData
2
p
 customKey="something"

<ns1:e11 xmlns:ns1="http://www.foo.com/ns1/">text1</ns1:e11><ns1:e11 xmlns:ns1="http://www.foo.com/ns1/">text2</ns1:e11>
<ns1:e12 xmlns:ns1="http://www.foo.com/ns1/"><![CDATA[cdata-section1]]></ns1:e12><ns1:e12 xmlns:ns1="http://www.foo.com/ns1/"><![CDATA[cdata-section2<&]]></ns1:e12>
<ns1:e1 xmlns:ns1="http://www.foo.com/ns1/" a1="v1" a2="v2">
    <ns1:e11>text1</ns1:e11>
    <ns1:e12><![CDATA[cdata-section1]]></ns1:e12>
  </ns1:e1>
<ns1:e11 xmlns:ns1="http://www.foo.com/ns1/">text1</ns1:e11><ns1:e11 xmlns:ns1="http://www.foo.com/ns1/">text2</ns1:e11>
a1
v2
rootroot
root
  root
  e1
  e11
  e12
  e2
  e11
  e12
  root
  root
  e1
  root
  e1
  e11
  root
  e1
  e12
  root
  e2
  root
  e2
  e11
  root
  e2
  e12
cdata-section2&lt;&amp;
cdata-section2<&
